src/odoc/resolver.ml

```diff
@@ -465,40 +542,64 @@ type roots = {
   lib_roots : (string * Fs.Directory.t) list;
   current_lib : string option;
   current_package : string option;
+  current_dir : Fs.Directory.t;
 }
 
 let create ~important_digests ~directories ~open_modules ~roots =
   let ap = Accessible_paths.create ~directories in
-  let pages, libs =
+  let pages, libs, hierarchy =
     match roots with
-    | None -> (None, None)
-    | Some { page_roots; lib_roots; current_lib; current_package } ->
-        ( Some (Named_roots.create ~current_root:current_package page_roots),
-          Some (Named_roots.create ~current_root:current_lib lib_roots) )
+    | None -> (None, None, None)
+    | Some { page_roots; lib_roots; current_lib; current_package; current_dir }
+      ->
+        let pages = Named_roots.create ~current_root:current_package page_roots
+        and libs = Named_roots.create ~current_root:current_lib lib_roots in
+        let hierarchy =
+          match Named_roots.find_by_path pages ~path:(Fpath.v ".") with
```

Comment:
Thanks, by using a bug you uncovered its existence!

As I though it, `find_by_path` was supposed to only return files (and not directories).
I think it is fine for it to return directories, but in this case we need to check that the output is a file (using `Sys.is_regular_file f` or `not (Sys.is_directory f)`)

Reply:
Here we want a directory. This requires a path that will be used to resolve relative references later, it is not expected to be a page.

Elsewhere, the lookup function loads the page/unit files before returning a positive result so no need to check that at any place. The lookup can't succeed on a directory.